### PR TITLE
docs - add info about greenplum database release versioning

### PR DESCRIPTION
**Pull request by lisakowen**
add new section to administrator guide describing components of greenplum database release version numbers.

doc review site link:
- http://docs-gpdb-review-staging.cfapps.io/540/admin_guide/managing/versioning.html

### CODE REVIEW

**Comment by mkiyama** [comment]
just wondering, should it be mentioned that some components have their own version, like GPORCA, or would it just be confusing at this level?

I've seen issue come up when building with mismatched versions.

**Comment by lisakowen** [comment]
this PR is for end user docs, the product should already be built with the appropriate version of gporca.

that sort of info would be useful in a readme or similar for developers if it does not already exist.

**Comment by dyozie** [comment]
Yeah - there also seems to be a desire to consolidate all those components
around the same database version number eventually (even gpcc/wlm, when
their releases get merged with the core db).  Especially for the Pivotal
product, I think we can ignore describing other component version
conventions for now.

File: gpdb-doc/dita/admin_guide/admin_guide.ditamap [hunk 1]
```diff
@@ -15,6 +15,7 @@
             <topicref href="intro/about_statistics.xml"/>
         </topicref>
         <topicref href="managing/partII.xml" id="partII">
+            <topicref href="managing/versioning.xml" navtitle="About Your Greenplum Database Release Version Number"/>
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/admin_guide.ditamap]: I'm open to feedback here, but I think having "Your" in this title is a little too colloquial.  Seems like "About the Greenplum Database ..." works better.

File: gpdb-doc/dita/admin_guide/managing/versioning.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  <body>
+    <p>A Greenplum Database release version number takes the format
+      <varname>x.y.z</varname>, where:
+      <ul>
+        <li><varname>x</varname> identifies the Major version number</li>
+        <li><varname>y</varname> identifies the Minor version number</li>
+        <li><varname>z</varname> identifies the Patch version number</li>
+      </ul>
+    </p>
+    <p>Greenplum Database releases with the same Major release number are guaranteed
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/managing/versioning.xml]: with -> having

(or "that have")

File: gpdb-doc/dita/admin_guide/managing/versioning.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      <varname>x.y.z</varname>, where:
+      <ul>
+        <li><varname>x</varname> identifies the Major version number</li>
+        <li><varname>y</varname> identifies the Minor version number</li>
+        <li><varname>z</varname> identifies the Patch version number</li>
+      </ul>
+    </p>
+    <p>Greenplum Database releases with the same Major release number are guaranteed 
+      to be backwards compatible.</p>
+    <p>Greenplum Database increments the Major release number when the catalog
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/managing/versioning.xml]: consolidate this with the previous paragraph.

File: gpdb-doc/dita/admin_guide/managing/versioning.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <li><varname>x</varname> identifies the Major version number</li>
+        <li><varname>y</varname> identifies the Minor version number</li>
+        <li><varname>z</varname> identifies the Patch version number</li>
+      </ul>
+    </p>
+    <p>Greenplum Database releases with the same Major release number are guaranteed 
+      to be backwards compatible.</p>
+    <p>Greenplum Database increments the Major release number when the catalog 
+      changes or when incompatible feature changes or new features are introduced.</p>
+    <p>The Minor release number for a given Major release is incremented when
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/managing/versioning.xml]: is incremented -> increments

File: gpdb-doc/dita/admin_guide/managing/versioning.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        <li><varname>z</varname> identifies the Patch version number</li>
+      </ul>
+    </p>
+    <p>Greenplum Database releases that have the same Major release number are 
+      guaranteed to be backwards compatible. Greenplum Database increments the
+      Major release number when the catalog changes or when incompatible feature
+      changes or new features are introduced.</p>
+    <p>The Minor release number for a given Major release increments when 
+      backwards compatible new features are introduced or when a Greenplum Database
+      feature is deprecated.</p>
```
Comment (danielgustafsson) [inline gpdb-doc/dita/admin_guide/managing/versioning.xml]: Feature deprecation is by definition a not backwards compatible change, so I would assume we only do that for major releases?

Comment (dyozie) [reply gpdb-doc/dita/admin_guide/managing/versioning.xml]: Actually, I'm not sure that feature deprecation (that is, adding a deprecation notice) has any correlation to product versioning.  It's probably more accurate to say that only a major release might *remove* a previously-deprecated feature.

**Comment by danielgustafsson** [comment]
> On 02 Jan 2018, at 17:51, David Yozie <notifications@github.com> wrote:
> 
> @dyozie commented on this pull request.

> Actually, I'm not sure that feature deprecation (that is, adding a deprecation notice) has any correlation to product versioning. It's probably more accurate to say that only a major release might remove a previously-deprecated feature.

Yes. Typically a feature is marked as deprecated during the development cycle of the next major release, but that’s rarely (if ever) backported. It’s better to reword it to say that previously deprecated functionality may be removed in a major release, but never in a minor release.
